native/src/hocs/withPayloadProvider.tsx

```diff
@@ -66,20 +67,28 @@ export type PropsType<
 > = StatusPropsType<S, R> & {
   dispatch: Dispatch<StoreActionType>
   navigation: NavigationPropType<T>
+  route: RoutePropType<T>
 }
 
 const withPayloadProvider = <
-  S extends { dispatch: Dispatch<StoreActionType> },
+  S extends {
+    dispatch: Dispatch<StoreActionType>
+  },
   R extends Record<string, unknown>,
   T extends RoutesType
 >(
   refresh: (refreshProps: R, dispatch: Dispatch<StoreActionType>) => void,
+  clearRouteOnClose: boolean,
```

Comment:
Why is this needed now?

Reply:
Because not all routes need to clear the route from redux on close. See the changes from #457, before a function `onRouteClose` was passed wherever necessary, but since the function is always the same (just dispatch an action) we don't need to pass it and a boolean is enough.